pypeit/spectrographs/gemini_gmos.py

```diff
@@ -482,10 +489,213 @@ def allowed_mosaics(self):
+            :attr:`slitmask`.
+        """
+        # Open the file
+        mask_tbl = Table.read(filename)
+
+        # Projected distance (in arcsec) of the object from the left and right (top and bot) edges of the slit
+        slit_length = mask_tbl['slitsize_y'].to('arcsec').value # arcsec
+        topdist = np.round(slit_length/2. - 
+                           mask_tbl['slitpos_y'].to('arcsec').value, 3)
+        botdist = np.round(slit_length/2. + 
+                           mask_tbl['slitpos_y'].to('arcsec').value, 3)
+
+        # Coordinates
+        # WARNING -- GMOS TABLE IS ONLY IN FLOAT32!!!
+        obj_ra = mask_tbl['RA'].value * 15.  
+        obj_dec = mask_tbl['DEC'].value 
+        objname = mask_tbl['ID'].value.astype(str)
+
+        slitID = mask_tbl['ID'].value # Slit and objects are the same
+
+        #   - Pull out the slit ID, object ID, name, object coordinates, top and bottom distance
+        objects = np.array([np.array(slitID, dtype=int),
+                           np.zeros(slitID.size, dtype=int),   # no object ID
+                           obj_ra,
+                           obj_dec,
+                           objname,
+                           np.array(mask_tbl['MAG'].value, dtype=float),
+                           ['None']*slitID.size,       # no magnitude band
+                           topdist,
+                           botdist]).T
+        # Mask pointing
+        mask_coord = SkyCoord(mask_tbl.meta['RA_IMAG'], mask_tbl.meta['DEC_IMAG'], 
+                              unit=("hourangle", "deg"))
+
+        # PA corresponding to positive x on detector (spatial)
+        posx_pa = mask_tbl.meta['MASK_PA'] - 180. # deg
+        if posx_pa < 0.:
+            posx_pa += 360.
+
+        # Slit positions
+        obj_coord = SkyCoord(ra=obj_ra, dec=obj_dec, unit='deg')
+        offsets = np.sqrt(
+                mask_tbl['slitpos_x'].to('arcsec').value**2 + 
+                mask_tbl['slitpos_y'].to('arcsec').value**2) 
+        # NOT READY FOR TILTS
+        if np.any(np.invert(np.isclose(mask_tbl['slittilt'].value, 0.))):
+            msgs.error('NOT READY FOR TILTED SLITS')
+        # NOT SURE WE HAVE THE TILT SIGN CORRECT
+        slit_pas = posx_pa + mask_tbl['slittilt'].to('deg').value
+        off_signs = np.ones_like(slit_pas)
+        negy = mask_tbl['slitpos_y'] < 0.
+        off_signs[negy] = -1.
+
+        slit_ra, slit_dec = [], []
+        for offset, coord, slit_pa, off_sign in zip(offsets, obj_coord, slit_pas, off_signs):
+            slit_coord = coord.directional_offset_by(
+                slit_pa*units.deg, off_sign*offset*units.arcsec)
+            slit_ra.append(slit_coord.ra.deg)
+            slit_dec.append(slit_coord.dec.deg)
+            
+
+        # Instantiate the slit mask object and return it
+        self.slitmask = SlitMask(
+            np.array(
+            [np.zeros(slitID.size),   # gemini_gmos maskdef has not slit corners
```

Comment:
Should we do some consolidation here?  I.e., if you don't need the corners for Gemini/GMOS, do we need them for the other spectrographs that use the mask design information?